From 1b331826d51c77aa5515478192fa0004ec93d887 Mon Sep 17 00:00:00 2001
From: Roman Stratiienko <r.stratiienko@gmail.com>
Date: Fri, 7 Apr 2023 11:05:02 +0300
Subject: [PATCH 1/3] Add dav1d project

https://code.videolan.org/videolan/dav1d

The idea for using dav1d in Android was inspired by:
https://github.com/raspberry-vanilla project.

Signed-off-by: Roman Stratiienko <r.stratiienko@gmail.com>
---
 Android.mk     |  1 +
 meson_dav1d.mk | 38 ++++++++++++++++++++++++++++++++++++++
 2 files changed, 39 insertions(+)
 create mode 100644 meson_dav1d.mk

diff --git a/Android.mk b/Android.mk
index f367251..3761b82 100644
--- a/Android.mk
+++ b/Android.mk
@@ -8,6 +8,7 @@ LOCAL_PATH := $(call my-dir)
 
 include $(LOCAL_PATH)/cargo_mmradio.mk
 
+include $(LOCAL_PATH)/meson_dav1d.mk
 include $(LOCAL_PATH)/meson_dbus.mk
 include $(LOCAL_PATH)/meson_glib.mk
 include $(LOCAL_PATH)/meson_libgudev.mk
diff --git a/meson_dav1d.mk b/meson_dav1d.mk
new file mode 100644
index 0000000..026e8fe
--- /dev/null
+++ b/meson_dav1d.mk
@@ -0,0 +1,38 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# AOSPEXT project (https://github.com/GloDroid/aospext)
+#
+# Copyright (C) 2021 GlobalLogic Ukraine
+# Copyright (C) 2021-2023 Roman Stratiienko (r.stratiienko@gmail.com)
+
+AOSPEXT_PROJECT_NAME := DAV1D
+
+ifneq ($(filter true, $(BOARD_BUILD_AOSPEXT_DAV1D)),)
+
+LOCAL_PATH := $(call my-dir)
+
+include $(CLEAR_VARS)
+
+LOCAL_SHARED_LIBRARIES := libc
+MESON_GEN_PKGCONFIGS :=
+
+MESON_BUILD_ARGUMENTS := \
+
+# Format: TYPE:REL_PATH_TO_INSTALL_ARTIFACT:VENDOR_SUBDIR:MODULE_NAME:SYMLINK_SUFFIX
+# TYPE one of: lib, bin, etc
+AOSPEXT_GEN_TARGETS := \
+    bin:dav1d::dav1d:                      \
+    lib:libdav1d.so::libdav1d:             \
+    $(BOARD_DAV1D_EXTRA_TARGETS)
+
+LOCAL_MULTILIB := first
+include $(LOCAL_PATH)/meson_cross.mk
+LOCAL_MULTILIB := first
+AOSPEXT_TARGETS_DEP:=$(MESON_GEN_FILES_TARGET)
+AOSPEXT_PROJECT_INSTALL_DIR:=$(dir $(AOSPEXT_TARGETS_DEP))/install
+AOSPEXT_PROJECT_OUT_INCLUDE_DIR:=$(AOSPEXT_PROJECT_INSTALL_DIR)/vendor/include
+include $(LOCAL_PATH)/aospext_gen_targets.mk
+
+#-------------------------------------------------------------------------------
+
+endif # BOARD_BUILD_AOSPEXT_DAV1D
-- 
2.37.2

